experimental/Schemes/CartierDivisor.jl

```diff
@@ -174,34 +174,60 @@ function cartier_divisor(IP::AbsProjectiveScheme, f::Union{MPolyDecRingElem, MPo
   return one(ZZ)*effective_cartier_divisor(IP, f)
 end
 
+### Decompostion of an effective Cartier Divisor into irreducible components
+### (specialized variant of associated_points, using pure codimension 1
+###  and taking multiplicities into account)
+@doc raw"""
+    irreducible_decomposition(C::EffectiveCartierDivisor)
+
+Return a `Vector` of pairs ``(I,k)`` corresponding to the irreducible components of ``C``. More precisely,  each ``I`` is a prime  `IdealSheaf` corresponding to an irreducible component of ``C`` and ``k``is the multiplicity of this component in ``C``.
+"""
+function irreducible_decomposition(C::EffectiveCartierDivisor)
+  cov = trivializing_covering(C)
+  X = scheme(C)
+  OOX = OO(X)
+
+  charts_todo = copy(patches(cov))
+  I = ideal_sheaf(C)
+  associated_primes_temp = Vector{Tuple{IdealSheaf, Int}}()  ## already identified components
+
+  # run through all charts and collect further irreducible components
+  while length(charts_todo) > 0
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                                ## supp(C) might not meet all charts
+    I_temp=I(U)
+
+    for (J,_) in associated_primes_temp
+      !is_one(J(U)) || continue
+      I_temp=saturation(I_temp,J(U))                         ## kick out known components
+      !is_one(I_temp) || break                               ## break if nothing left
+    end
+
+    !is_one(I_temp) || break                                 ## break if nothing left
+    components_here = minimal_primes(I(U))
+    for comp in components_here
+      I_temp, saturation_index = saturation_with_index(I_temp, comp)
+      temp_dict=IdDict{AbsSpec,Ideal}()
+      temp_dict[U] = comp
+      I_sheaf_temp = IdealSheaf(X, extend!(cov, temp_dict), check=false)
```

Comment:
Is this line tested? Frankly, I do not understand what's going on. It seems you're extending a given covering by a dictionary. I don't recall having implemented such functionality and I can't really think of what this should mean. Could you explain what you intend to do here?

Reply:
We are extending the dictionary to the covering, not the covering by a dictionary. (we are dealing with irreducible codimension 1 subschemes, so there is no guesswork going on here).
This is functionality which you provide in IdealSheaves.jl and which is documented and tested (not by me..)

Reply:
But you are right: 
I could have written 
            IdealSheaf(X,U,gens(comp)) 
but would not have had the option to swith off the tests there (which are true by construction here). I have borrowed this line from your code there.

Reply:
Oh, je... It was me then? At least it seems to have been documented...